src/applications/MachineShopSimulator.java

```diff
@@ -32,49 +32,16 @@ static boolean moveToNextMachine(Job theJob, SimulationResults simulationResults
                 // get machine for next task
             int p = ((Task) theJob.getTaskQ().getFrontElement()).getMachine();
             // put on machine p's wait queue
-            machine[p].getJobQ().put(theJob);
+            machine[p].putJobOnQ(theJob);
             theJob.setArrivalTime(timeNow);
             // if p idle, schedule immediately
             if (eList.nextEventTime(p) == largeTime) {// machine is idle
-                changeState(p);
+                machine[p].changeState(timeNow);

```

Comment:
Now this class looks cleaner and more readable. It definitely makes more sense to have changeState() method in the Machine class.